src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,668 @@
+        Specification : https://www.nordugrid.org/arc/arc6/tech/rest/rest.html
+        """
+        if hasattr(self, "s"):
+            # Initialisation has already been successfully ran
+            return True
+        self.log.debug("Testing if the REST interface is available", "for %s" % self.ceHost)
+
+        result = self._prepareProxy()
+        if not result["OK"]:  # Probably everything is going to fail?
+            self.log.info("No proxy found -  Probably just no jobs to run for this CE?")
+            self.log.info("Setting REST interface false.")
+            return False
+
+        # Get the REST endpoint (service_url in ARC language) from the CS if available (preferred).
+        # It should be at
+        #   Resources/Sites/<Grid>/<Site>/CEs/<CE>/RESTEndpoint
+        # Otherwise query the CE for this value.
+
+        # Try getting service_url from the CS
+        ceString = cfgPath("Resources", "Sites", self.grid, self.site, "CEs", self.ceHost, "RESTEndpoint")
+        result = self._getFromCfgSvc(ceString)
+        if result["OK"]:
+            service_url = result["Value"]  # All well
+        if len(service_url) < 5:  # There is no endpoint in the CS. Discover it.
+            # The following command should expand to for example
+            # ldapsearch -x -LLL -h grendel.hec.lancs.ac.uk:2135 -b 'o=glue' GLUE2EndpointInterfaceName=org.nordugrid.arcrest GLUE2EndpointURL | grep GLUE2EndpointURL | awk -F ": " '{print $2}'
+            server = ldap3.Server("ldap://" + self.ceHost + ":2135")
+            try:
+                connection = ldap3.Connection(server, client_strategy=ldap3.SAFE_SYNC, auto_bind=True)
+            except ldap3.core.exceptions.LDAPSocketOpenError:
+                self.log.error("Could not connect to server", "CE : %s:2135" % self.ceHost)
+                return False
+            status, result, response, _ = connection.search(
+                "o=glue", "(GLUE2EndpointInterfaceName=org.nordugrid.arcrest)", attributes="GLUE2EndpointURL"
+            )
+            if not status or len(response) == 0:
+                self.log.error("Bad status from LDAP search", result)
+                return False
+            if len(response) != 1:
+                self.log.warning("Expected one endpoint, got %s. Using the first." % len(response))
+            # For some reason the response is of class "bytes"
+            service_url = response[0]["raw_attributes"]["GLUE2EndpointURL"][0].decode()
+
+        # We now have a pointer (service_url) to the REST interface for this CE
+        restVersion = "1.0"  # Will be this value for the forseeable future.
+        self.base_url = service_url + "/rest/" + restVersion + "/"
+
+        # Set up the request framework
+        self.session = requests.Session()
+        self.session.verify = Locations.getCAsLocation()
+        self.headers = {"accept": "application/json", "Content-Type": "application/json"}
```

Comment:
Done - thanks.

Reply:
One possible point is that the "self.session" is now defined in the function "initialised" instead of "__init__". 

From formally pure python, I am guessing this is not very elegant. Any thoughts as to how I can tell that the initialisation has successfully run without looking for "session" in "self"?

Reply:
`__init__` will only be called once.

Reply:
Superseded by the changes made by @aldbr